.grit/patterns/js/prototype_to_class.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Rewrite `Function.prototype.method = ` ⇒ es6 class
+---
+
+Older code often uses the function prototype to create "classes" out of functions. This upgrades those to ES6 class syntax.
+
+tags: #ES6
+
+```grit
+engine marzano(0.1)
+language js
+
+pattern prototype_to_class() {
+    program(statements = $statements) where {
+        $statements <: contains bubble($statements) function_declaration(body=$body, name=$class, parameters=$args) as $root where {
```

Comment:
This approach is causing a buffer overflow, likely due to autowrapping the root pattern.

Instead you can just directly find it:

```suggestion
pattern prototype_to_class() {
    function_declaration(body=$body, name=$class, parameters=$args) as $root where {
```

Then when you need to search for other elements in the program just match against `$program`, which is always globally bound.

Reply:
Ohhh, thanks! I didn't know about `$program` being available. I can change a lot of my other scripts to use this too I think.